exo/networking/manual/manual_discovery.py

```diff
@@ -60,12 +64,65 @@ async def task_find_peers_from_config(self):
             self.known_peers[peer_id] = peer
           else:
             if DEBUG_DISCOVERY >= 2: print(f"{peer_id=} at {peer_config.address}:{peer_config.port} is not healthy.")
-            try:
-              del self.known_peers[peer_id]
-            except KeyError:
-              pass
+            self.known_peers.pop(peer_id, None)
         except Exception as e:
           if DEBUG_DISCOVERY >= 2: print(f"Exception occured when attempting to add {peer_id=}: {e}")
       await asyncio.sleep(1.0)
 
       if DEBUG_DISCOVERY >= 2: print(f"Current known peers: {[peer.id() for peer in self.known_peers.values()]}")
+
+  async def task_clean_up_peers_from_config(self):
+    if DEBUG_DISCOVERY >= 2: print("Starting task to clean up peers from config...")
+    while True:
+      peers_from_config = await self._get_peers()
+      if peers_from_config:
+        peers_to_remove = [peer for peer in self.known_peers.keys() if peer not in peers_from_config]
+
+        for peer in peers_to_remove:
+          if DEBUG_DISCOVERY >= 2: print(f"{peer} is no longer found in the config but is currently a known peer. Removing from known peers...")
+          self.known_peers.pop(peer, None)
+
+      await asyncio.sleep(1.0)
```

Comment:
I'm not sure why this is needed. I think this may be overcomplicating it.

The way `StandardNode` is set up, it should be possible to simply return the latest set of peers, i.e. your `_get_peers` is enough.

Reply:
Ah I see, TIL a bit more about StandardNode, thanks!

I thought it was necessary to handle cleaning up nodes that were originally added to the config file (and thus added to `self.known_peers`) but removed afterwards.